lib/usd/utils/MergePrimsOptions.h

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsdUtils/Api.h>
+
+#include <pxr/base/tf/staticTokens.h>
+#include <pxr/base/tf/token.h>
+#include <pxr/base/vt/dictionary.h>
+
+namespace MayaUsdUtils {
+
+//----------------------------------------------------------------------------------------------------------------------
+/// MergeVerbosity level flags.
+
+enum class MergeVerbosity
+{
+    None = 0,
+    Same = 1 << 0,
+    Differ = 1 << 1,
+    Child = 1 << 2,
+    Children = 1 << 3,
+    Failure = 1 << 4,
+    Default = Differ | Child | Children | Failure,
+};
+
+inline MergeVerbosity operator|(MergeVerbosity a, MergeVerbosity b)
+{
+    return MergeVerbosity(uint32_t(a) | uint32_t(b));
+}
+
+inline MergeVerbosity operator&(MergeVerbosity a, MergeVerbosity b)
+{
+    return MergeVerbosity(uint32_t(a) & uint32_t(b));
+}
+
+inline MergeVerbosity operator^(MergeVerbosity a, MergeVerbosity b)
+{
+    return MergeVerbosity(uint32_t(a) ^ uint32_t(b));
+}
+
+inline bool contains(MergeVerbosity a, MergeVerbosity b) { return (a & b) != MergeVerbosity::None; }
+
+//----------------------------------------------------------------------------------------------------------------------
+// Missing field handling flags.
+
+enum class MergeMissing
+{
+    None = 0,
+
+    // If set, attributes found only in the source are created in the destination.
+    Create = 1 << 0,
+
+    // If set, attributes missing from the source are preserved in the destination.
+    Preserve = 1 << 1,
+
+    All = Create | Preserve,
+};
+
+inline MergeMissing operator|(MergeMissing a, MergeMissing b)
+{
+    return MergeMissing(uint32_t(a) | uint32_t(b));
+}
+
+inline MergeMissing operator&(MergeMissing a, MergeMissing b)
+{
+    return MergeMissing(uint32_t(a) & uint32_t(b));
+}
+
+inline MergeMissing operator^(MergeMissing a, MergeMissing b)
+{
+    return MergeMissing(uint32_t(a) ^ uint32_t(b));
+}
+
+inline bool contains(MergeMissing a, MergeMissing b) { return (a & b) != MergeMissing::None; }
+
+//----------------------------------------------------------------------------------------------------------------------
+// Options to control prims merging.
+
+struct MergePrimsOptions
+{
+    // How much logging is done during the merge.
+    MergeVerbosity verbosity = MergeVerbosity::Default;
+
+    // if true, merges children too, otherwise merge only the given prim.
+    bool mergeChildren = false;
```

Comment:
Elsewhere, for structs initialized from dictionaries, we're making all the data members const, e.g.
https://github.com/Autodesk/maya-usd/blob/2bf98c7f2fae04d6237a43c8d060e6c065e589e1/lib/mayaUsd/fileio/jobs/jobArgs.h#L150
This certainly reinforces the notion that these objects are pure values that cannot be mutated.  Now I see that you have a public default constructor, which allows you to later on to initialize your choice of the non-const data members, so having them non-const is more flexible for initialization.  If you choose to leave them non-const, would be good to put in a single-line comment explaining why.

Reply:
I expected the option to be kept around, but I can see if making them const breaks anything.

Reply:
Be careful with this, you've changed test_MergePrims.cpp to use non-constness for initialization purposes.  Don't know if it's worth your time, if your investigation takes you more than 10 minutes I'd just comment and move on.

Reply:
True, for testing having to pass through a dict just to set variable would be awkward.